host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java

```diff
@@ -443,12 +443,16 @@ public void operationPrepared(final ModelController.OperationTransaction transac
             if(failed) {
                 transaction.rollback();
             } else {
-                try {
-                    registerHost(transaction, result);
-                } catch (SlaveRegistrationException e) {
-                    failed(e, e.getErrorCode(), e.getErrorMessage());
-                } catch (Exception e) {
-                    failed(e, SlaveRegistrationException.ErrorCode.UNKNOWN, e.getClass().getName() + ":" + e.getMessage());
+                if (this.extensionRegistry.enables(this.hostInfo)) {
+                    try {
+                        registerHost(transaction, result);
+                    } catch (SlaveRegistrationException e) {
+                        failed(e, e.getErrorCode(), e.getErrorMessage());
+                    } catch (Exception e) {
+                        failed(e, SlaveRegistrationException.ErrorCode.UNKNOWN, e.getClass().getName() + ":" + e.getMessage());
+                    }
+                } else {
+                    this.failed(SlaveRegistrationException.ErrorCode.INCOMPATIBLE_VERSION, DomainControllerLogger.ROOT_LOGGER.hostFeatureStreamNotEnabled(this.extensionRegistry.getFeatureStream(), this.hostInfo.getFeatureStream()));
```

Comment:
Hi @pferraro , the rejection should be done a bit earlier, more specifically at https://github.com/wildfly/wildfly-core/blob/main/host-controller/src/main/java/org/jboss/as/host/controller/mgmt/HostControllerRegistrationHandler.java#L336-L348

You can get a reference of the ExtensioRegistry using the HostControllerRegistrationHandler constructor.
That's also the place where we reject is a host is incompatible due to its kernel version and we know we can reject it there.

Reply:
@yersan Makes sense.  Will fix.

Reply:
@yersan I've updated that commit.  Let me know if it looks OK now.

Reply:
Thank you, @pferraro LGTM